src/bridge/cockpitdbusjson.c

```diff
@@ -2622,6 +2635,15 @@ cockpit_dbus_json_init (CockpitDBusJson *self)
 
   self->peers = g_hash_table_new (g_str_hash, g_str_equal);
   self->interface_info = cockpit_dbus_interface_info_new ();
+
+  /* create global objects */
+  if (group_connections == NULL)
+    {
+      group_connections = g_hash_table_new_full (g_str_hash, g_str_equal,
+                                                 g_free, g_object_unref);
+      pending_group_connections = g_hash_table_new_full (g_str_hash, g_str_equal,
+                                                         g_free, (GDestroyNotify) g_ptr_array_unref);
```

Comment:
It seems like the cockpit_dbus_json_finalize() function should check if these are empty and free/NULL them if so. Cleaning up memory like this helps us track down other leaks/issues.